Lib/subprocess.py

```diff
@@ -978,6 +979,18 @@ def __init__(self, args, bufsize=-1, executable=None,
 
             raise
 
+    def __repr__(self):
+        max_args_length = 80
+        args = ' '.join(map(shlex.quote, self.args))
+
+        if len(args) > max_args_length:
+            args = f"{args[:max_args_length-3]}..."
+
+        return (
+            f"<{self.__class__.__name__}: "
+            f"returncode:'{self.returncode}' args:'{args}'>"
```

Comment:
I think the return code should not be surrounded by quotes: it will be either an integer or `None`, but never a string.

I'm not sure about args, but if we go with putting quotes around it, that would be interpreted as being a string literal, so we should use its repr, i.e. `args:{args!r}`.

Reply:
I agree regarding the return code.

Also regarding the args: I suggested `map(shlex.quote, args)` but I know there's also `subprocess.list2cmdline` which has different results. I picked the former arbitrarily, if you have reason to prefer the latter, go for it. 

In any case, @dorosch should definitely remove the quotes he included and use `!r` as you suggested.

Reply:
Note that `list2cmdline()` is only for use on Windows!!

It may be wise to use it if `_mswindows` is true, falling back to `' '.join(map(shlex.quote, args))` otherwise.

Or just print the args as a list and avoid the whole issue:
```python3
f'<Popen: ... args: {list(self.args)!r}>
```

Reply:
> Note that `list2cmdline()` is only for use on Windows!!

That was worth knowing! Thanks.

In that case, my recommendation is to stay with the `shlex` code that's in the PR. I'd avoid printing the args as a list, it's always annoying to see these lists when debugging. (Of course, they should always be used when calling subprocess functions.)

Reply:
Yes, command lines are easier to read when not written as lists. On the other hand, a list matches the interface of the `Popen()` constructor and the internal representation, is not platform-dependent, and is unambiguous.

It's a good sign that everything is settling down and we're left just with "bike-shedding" ;) The proper place for this would be the bug tracker. Care to bring this up there, @cool-RR?

Reply:
Nah, it's indeed bikeshedding and we can let it go. After removing the shlex imports, this PR is good to go.